src/main/java/me/qoomon/gitversioning/GitVersioning.java

```diff
@@ -86,4 +93,21 @@ public static GitVersionDetails determineVersion(
                 gitVersion
         );
     }
+
+    private static String dateTimeOrEmpty(Date headCommitDate){
+        if (headCommitDate == null){
+            return "";
+        }
+        return DateTimeFormatter
+                .ofPattern(VERSION_DATE_TIME_FORMAT)
+                .withZone(ZoneOffset.UTC)
+                .format(headCommitDate.toInstant());
+    }
+
+    private static String timestampOrEmpty(Date headCommitDate){
+        if (headCommitDate == null){
+            return "";
```

Comment:
I would prefer to return "0" instead of "" because an empty string would look strange in a version pattern like this "{branch}-{commit.timestamp}"